src/ipc/ipc-helper.c

```diff
@@ -180,19 +179,25 @@ int comp_verify_params(struct comp_dev *dev, uint32_t flag,
 int comp_buffer_connect(struct comp_dev *comp, uint32_t comp_core,
 			struct comp_buffer *buffer, uint32_t buffer_core, uint32_t dir)
 {
+	struct comp_dev *cd = buffer_get_comp(buffer, dir);
 	int ret;
 
 	/* check if it's a connection between cores */
-	if (buffer_core != comp_core) {
+	if (cd && cd->ipc_config.core != comp_core) {
 		dcache_invalidate_region(buffer, sizeof(*buffer));
-
 		buffer->inter_core = true;
 
 		if (!comp->is_shared) {
 			comp = comp_make_shared(comp);
 			if (!comp)
 				return -ENOMEM;
 		}
+
+		if (!cd->is_shared) {
+			cd = comp_make_shared(cd);
+			if (!cd)
+				return -ENOMEM;
```

Comment:
`comp_make_shared()` cannot fail. I'd ignore the return value unless we have to use it. Also maybe remove the similar check above for `comp`.